### PR TITLE
Create codeql.yml

### PR DESCRIPTION
**Pull request by Jimexist**
## Which issue does this PR close?

Closes #.

## Rationale for this change

so that we can get code scanned by GitHub for security issues

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'python' ]
+        # CodeQL supports [ 'cpp', 'csharp', 'go', 'java', 'javascript', 'python', 'ruby', 'swift' ]
```
Comment (alamb) [inline .github/workflows/codeql.yml:42]: 🤔  it seems like codeql does not yet support Rust: https://docs.github.com/en/code-security/code-scanning/introduction-to-code-scanning/about-code-scanning-with-codeql

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+jobs:
+  analyze:
+    name: Analyze
+    # Runner size impacts CodeQL analysis time. To learn more, please see:
+    #   - https://gh.io/recommended-hardware-resources-for-running-codeql
+    #   - https://gh.io/supported-runners-and-hardware-resources
+    #   - https://gh.io/using-larger-runners
+    # Consider using larger runners for possible analysis time improvements.
+    runs-on: ${{ (matrix.language == 'swift' && 'macos-latest') || 'ubuntu-latest' }}
```
Comment (alamb) [inline .github/workflows/codeql.yml:31]: This language doesn't look quite right to me -- I don't think we have any swift code in DataFusion

**Comment by alamb** [comment]
Thanks for the contributions @Jimexist  but I am not quite sure this is configuring CodeQL correctly

**Comment by Jimexist** [comment]
@alamb yes you are right - i only added this with autogenerated code. maybe we can close this.

**Pull request closed without merging**